core/io/image_loader.cpp

```diff
@@ -82,28 +82,36 @@ Error ImageLoader::load_image(String p_file, Ref<Image> p_image, FileAccess *p_c
 
 void ImageLoader::get_recognized_extensions(List<String> *p_extensions) {
 
-	for (int i = 0; i < loader_count; i++) {
+	for (int i = 0; i < loader.size(); i++) {
 
 		loader[i]->get_recognized_extensions(p_extensions);
 	}
 }
 
-bool ImageLoader::recognize(const String &p_extension) {
+ImageFormatLoader *ImageLoader::recognize(const String &p_extension) {
 
-	for (int i = 0; i < loader_count; i++) {
+	for (int i = 0; i < loader.size(); i++) {
 
 		if (loader[i]->recognize(p_extension))
-			return true;
+			return loader[i];
 	}
 
-	return false;
+	return NULL;
 }
 
-ImageFormatLoader *ImageLoader::loader[MAX_LOADERS];
-int ImageLoader::loader_count = 0;
+Vector<ImageFormatLoader *> ImageLoader::loader;
```

Comment:
This vector needs to be cleared at some point at exit, or its memory will be leaked (or Godot crash)

Reply:
@reduz 
Would you prefer having something like ImageLoader::cleanup() which does this (if so, where should it go?) 
OR
make it the responsibility of the calling code that calls ImageLoader::add_image_format_loader
to remove its image loader before it destroys the image loader
(I think this second option is safer)

Reply:
@reduz, this is now fixed - I've added calls to remove the image loader which will free the vector once it resizes itself to zero